pkg/controller/provider/validation.go

```diff
@@ -196,18 +196,28 @@ func (r *Reconciler) validateConnectionStatus(provider *api.Provider, secret *co
 			Category: Warn,
 			Message:  "TLS is susceptible to machine-in-the-middle attacks when certificate verification is skipped.",
 		})
-	} else {
-		_, err := base.VerifyTLSConnection(provider.Spec.URL, secret)
-		if err != nil {
-			provider.Status.SetCondition(libcnd.Condition{
-				Type:     ConnectionTestFailed,
-				Status:   True,
-				Reason:   Tested,
-				Category: Critical,
-				Message:  err.Error(),
-			})
-		}
+		return true
 	}
+
+	// Check if cacert is provided when TLS verification is enabled
+	if len(secret.Data["cacert"]) == 0 {
+		// Return false to signal caller to skip further validation (GetTlsCertificate)
+		// The keyList validation will handle the error
+		return false
```

Comment:
This should be separate function it's not realted to the validateConnectionStatus.